src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1578,6 +1586,20 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 ->addTag('messenger.receiver', array('alias' => $name))
                 ->addTag('messenger.sender', array('alias' => $name))
             ;
+
+            if ('symfony://kernel.terminate' === $transport['dsn']) {
+                $busId = $transport['options']['bus'] ?? null;
```

Comment:
That way you don't supports the query parameters, like this as a DSN:
```yaml
symfony://kernel.terminate?bus=commands
```

I think that this resolving of the bus should be in the transport itself, not here. The only issue is that it would throw an exception at runtime if the bus is wrong (and runtime here is after the view is rendered :/). So I think you should do the parsing of the DSN (+ options) as a static method on the transport and use it here (throw an exception if class does not exists).

Reply:
Does this as a DSN query param really make sense actually?
1. It's mandatory
2. I don't see the need for it to be set differently according to the environment. Which is what allows the DSN as it's likely to be stored in a env var / parameter.

Reply:
I'm with ogizanagi on this one

Reply:
Consistency is my main argument. As `options` or in the DSN, we need to manually enforce it to be mandatory 🙃

Reply:
@sroze: `MemoryTransport::send` will be called before `kernel.terminate`, so the middleware will be instantiated before the view is rendered. Which means the factory will throw on wrong bus before rendering. So we're fine doing it only in the factory :)
(we can keep a check in the extension if it was defined through the option rather than the DSN though, for compilation-time error)

Reply:
I might have misunderstood it, but isn’t @sroze saying that dsn should also be parsed here by calling a static method on the transport so that the exception is thrown during compilation?

Reply:
Yes, but as explained in the comment above, that's not really necessary.
Also probably not desirable regarding https://github.com/symfony/symfony/pull/28746#discussion_r229894861